lua/ulx/modules/sh/timed_darkrp.lua

```diff
@@ -5,19 +5,35 @@ local HELP = "Prevents the target(s) from spawning anything at all"
 
 if SERVER then
     local function enable( ply )
-        ply:DisallowNoclip( true )
-        ply:DisallowSpawning( true )
+        ply.isInULXDarkRP = true
 
         ply:StripWeapons()
         cleanup.CC_Cleanup( ply, "gmod_cleanup", {} )
     end
 
     local function disable( ply )
-        ply:DisallowNoclip( false )
-        ply:DisallowSpawning( false )
+        ply.isInULXDarkRP = nil
+    end
+
+    local function blockAction( ply )
+        if not ply or not ply:IsValid() then return end
+        if ply.isInULXDarkRP then return false end
     end
 
     TimedPunishments.Register( PUNISHMENT, enable, disable )
+
+    -- Block spawning and noclip while in darkrp mode.
+    -- Has to be done manually to prevent ulib's :DisallowSpawning() and etc from interfering, such as when a player unragdolls.
+    hook.Add( "PlayerNoClip", "ULibNoclipCheck", noclip, HOOK_HIGH )
```

Comment:
Should the name of the hook be changed? I also dont see where `noclip` is defined.